### ISSUE TITLE
Refactored Accuracy metric no longer tolerate empty tensor

### ISSUE THREAD

**Issue by kasakun** [Lightning-AI/torchmetrics#2279]
## 🐛 Bug

In the middle of upgrading from 0.8.1 to 1.2.1, I found `Accuracy` metric no longer takes empty tensor. 

### To Reproduce

Same code breaks with 1.2.1:

```
    metric = torchmetrics.Accuracy(
        task="multiclass",
        num_classes=3,
        ignore_index=2,
    )
    mask = torch.tensor([False, False])
    preds = torch.tensor([1, 1])
    target = torch.tensor([1, 1])
    acc = metric(preds[mask], target[mask])
```

It ends up here: https://github.com/Lightning-AI/torchmetrics/blob/v1.2.1/src/torchmetrics/functional/classification/stat_scores.py#L339

I think this may apply to other metrics as well, like binary accuracy, etc.

<details>
  <summary>Code sample</summary>

</details>

### Expected behavior

To keep the backward compatibility, IMO this should be a no-op.

### Environment

- TorchMetrics version (and how you installed TM, e.g. `conda`, `pip`, build from source):
- Python & PyTorch Version (e.g., 1.0):
- Any other relevant information such as OS (e.g., Linux):
PyTorch : 2.0
Python: 3.10
